File: c/cap4/cap4.part1ex01.c
/* Copyright (C) 2022-2023 Francisco de Assis Zampirolli - All Rights Reserved
 * Author: Francisco de Assis Zampirolli
 * Purpose: Examples and some exercises from the Information Processing book:
 * https://editora.ufabc.edu.br/matematica-e-ciencias-da-computacao/58-processando-a-informacao
 * Language: C
 */

#include <stdio.h>

int main(void) {

  // ENTRADA DE DADOS e PROCESSAMENTO
  float acumulador = 0, nota, media;
  int contador = 0;

  printf("Entre com 10 notas válidas\n");
  while (contador < 10) {
    do {
      scanf("%f", &nota);
    } while (nota < 0.0 || nota > 10.0);
    acumulador = acumulador + nota;
    contador++;
  }
  media = acumulador / contador;

  // SAÍDA
  printf("A média das %d notas é %.1f\n", contador, media);

  return 0;
}

File: py/cap4/cap4.part1ex01.py
"""
Copyright(C) 2022 - 2023 Francisco de Assis Zampirolli - All Rights Reserved
Author : Francisco de Assis Zampirolli
Purpose : Examples and some exercises from the Information Processing book
Language : Python
"""

# ENTRADA DE DADOS e PROCESSAMENTO
acumulador, contador = 0, 0
print("Entre com 10 notas válidas")
while contador < 10:
    nota = float(input())
    if nota < 0 or nota > 10:
        print("ERRO, nota inválida. Digite uma nota entre 0 e 10:")
        continue
    acumulador += nota
    contador += 1

media = acumulador / contador

# SAÍDA
print("A média das " + str(contador) + " notas é " + str(round(media, 1)))
